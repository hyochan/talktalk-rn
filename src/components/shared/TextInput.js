// @flow
import React, { useState } from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  Image,
  Text,
  View,
  TextInput,
} from 'react-native';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import { ratio, colors } from '../../utils/Styles';

type Styles = {
  wrapper: ViewStyle,
  label: ViewStyle,
  labelFocus: TextStyle,
  input: ViewStyle,
  inputFocus: TextStyle,
};

const styles: Styles = StyleSheet.create({
  wrapper: {
    alignSelf: 'stretch',

    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
  },
  label: {
    color: colors.blueyGray,
    marginBottom: 8,
    fontSize: 12,
  },
  labelFocus: {
    color: colors.dodgerBlue,
    marginBottom: 8,
    fontSize: 12,
  },
  input: {
    alignSelf: 'stretch',
    color: colors.dusk,
    fontSize: 16,
    paddingVertical: 22,
    paddingHorizontal: 20,
    borderWidth: 1,
    borderColor: 'rgb(233,237,244)',

    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  inputFocus: {
    alignSelf: 'stretch',
    color: colors.dusk,
    fontSize: 16,
    paddingVertical: 22,
    paddingHorizontal: 20,
    borderWidth: 1,
    borderColor: colors.dodgerBlue,

    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
});

type Props = {
  style?: ViewStyle;
  labelStyle?: TextStyle;
  labelStyleFocus?: TextStyle;
  placeholderTextColor?: string;
  inputStyle?: TextStyle;
  isPassword?: boolean;
  multiline?: boolean;
  txtLabel?: string;
  txtHint?: string;
  txt?: string;
  onFocus?: () => void;
  onTextChanged?: (text: string) => void;
  onSubmitEditing?: (text: string) => void;
  returnKeyType?: any;
};

type State = {
  focused: boolean;
};

function Shared(props: Props, state: State) {
  const [focused, setFocused] = useState(false);

  const renderTxtLabel = () => {
    if (props.txtLabel) {
      return (
        <Text style={focused ? props.labelStyleFocus : props.labelStyle}>
          {props.txtLabel}
        </Text>
      );
    }
    return null;
  };

  return (
    <View style={[
      styles.wrapper,
      props.style,
    ]}>
      {renderTxtLabel()}
      <TextInput
        style={[
          focused ? styles.inputFocus : styles.input,
          props.inputStyle,
        ]}
        underlineColorAndroid='transparent' // android fix
        autoCapitalize='none'
        autoCorrect={false}
        multiline={props.multiline}
        onChangeText={props.onTextChanged}
        value={props.txt}
        onFocus={() => setFocused(true)}
        onBlur={() => setFocused(false)}
        placeholder={props.txtHint}
        placeholderTextColor={props.placeholderTextColor}
        // onSubmitEditing={this.props.onSubmitEditing}
        returnKeyType={props.returnKeyType}
        secureTextEntry={props.isPassword}
      />
    </View>
  );
}

Shared.defaultProps = {
  style: styles.wrapper,
  labelStyle: styles.label,
  labelStyleFocus: styles.labelFocus,
};

export default Shared;
