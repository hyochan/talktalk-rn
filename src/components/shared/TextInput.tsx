import React, { useState } from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  Image,
  Text,
  View,
  TextInput,
  ViewStyle,
  TextStyle,
} from 'react-native';

import { ratio, colors } from '../../utils/Styles';

interface IStyles {
  wrapper: ViewStyle;
  label: ViewStyle;
  labelFocus: TextStyle;
  input: ViewStyle;
  inputFocus: TextStyle;
}

const styles: IStyles = StyleSheet.create({
  wrapper: {
    alignSelf: 'stretch',

    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
  },
  label: {
    color: colors.blueyGray,
    marginBottom: 8,
    fontSize: 12,
  },
  labelFocus: {
    color: colors.dodgerBlue,
    marginBottom: 8,
    fontSize: 12,
  },
  input: {
    alignSelf: 'stretch',
    color: colors.dusk,
    fontSize: 16,
    paddingVertical: 22,
    paddingHorizontal: 20,
    borderWidth: 1,
    borderColor: 'rgb(233,237,244)',

    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  inputFocus: {
    alignSelf: 'stretch',
    color: colors.dusk,
    fontSize: 16,
    paddingVertical: 22,
    paddingHorizontal: 20,
    borderWidth: 1,
    borderColor: colors.dodgerBlue,

    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
});

interface IProps {
  testID?: string;
  style?: ViewStyle;
  labelStyle?: TextStyle;
  labelStyleFocus?: TextStyle;
  placeholderTextColor?: string;
  inputStyle?: TextStyle;
  isPassword?: boolean;
  multiline?: boolean;
  txtLabel?: string;
  txtHint?: string;
  txt?: string;
  onFocus?: () => void;
  onTextChanged?: (text: string) => void;
  onSubmitEditing?: (text: string) => void;
  returnKeyType?: any;
}

interface IState {
  focused: boolean;
}

function Shared(props: IProps, state: IState) {
  const [focused, setFocused] = useState(false);

  const renderTxtLabel = () => {
    if (props.txtLabel) {
      return (
        <Text style={focused ? props.labelStyleFocus : props.labelStyle}>
          {props.txtLabel}
        </Text>
      );
    }
    return null;
  };

  return (
    <View style={[
      styles.wrapper,
      props.style,
    ]}>
      {renderTxtLabel()}
      <TextInput
        style={[
          focused ? styles.inputFocus : styles.input,
          props.inputStyle,
        ]}
        underlineColorAndroid='transparent' // android fix
        autoCapitalize='none'
        autoCorrect={false}
        multiline={props.multiline}
        onChangeText={props.onTextChanged}
        value={props.txt}
        onFocus={() => setFocused(true)}
        onBlur={() => setFocused(false)}
        placeholder={props.txtHint}
        placeholderTextColor={props.placeholderTextColor}
        // onSubmitEditing={this.props.onSubmitEditing}
        returnKeyType={props.returnKeyType}
        secureTextEntry={props.isPassword}
      />
    </View>
  );
}

Shared.defaultProps = {
  style: styles.wrapper,
  labelStyle: styles.label,
  labelStyleFocus: styles.labelFocus,
};

export default Shared;
