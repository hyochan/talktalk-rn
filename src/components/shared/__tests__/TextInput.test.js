import 'react-native';
import * as React from 'react';
import TextInput from '../TextInput';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';

const component: React.Element<any> = <TextInput />;

describe('[TextInput] rendering test', () => {
  it('renders [enabled] as expected', () => {
    const json = renderer.create(component).toJSON();
    expect(json).toMatchSnapshot();
  });
});
