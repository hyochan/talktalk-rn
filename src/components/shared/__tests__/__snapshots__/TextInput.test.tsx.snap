// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[TextInput] rendering test renders [enabled] as expected 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "alignSelf": "stretch",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
      Object {
        "alignItems": "flex-start",
        "alignSelf": "stretch",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    onBlur={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "borderColor": "rgb(233,237,244)",
          "borderWidth": 1,
          "color": "rgb(65,77,107)",
          "flexDirection": "row",
          "fontSize": 16,
          "justifyContent": "flex-start",
          "paddingHorizontal": 20,
          "paddingVertical": 22,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
